src/Compilers/CSharp/Portable/Syntax/InternalSyntax/SyntaxList`1.cs

```diff
@@ -9,10 +9,12 @@
 using Microsoft.CodeAnalysis.Text;
 using Roslyn.Utilities;
 using System.Diagnostics;
+using Microsoft.CodeAnalysis.Syntax.InternalSyntax;
 
 namespace Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax
 {
-    internal partial struct SyntaxList<TNode> : IEquatable<SyntaxList<TNode>> where TNode : CSharpSyntaxNode
+    internal partial struct SyntaxList<TNode> : IEquatable<SyntaxList<TNode>>, ISyntaxList<TNode> 
+        where TNode : CSharpSyntaxNode
```

Comment:
Perhaps we should be looking at the final state then?
Such changes often introduce semi-unified state that is worse (slower), before making things better again.